### PR TITLE
Feature/session provider listener

### PR DESCRIPTION
**Pull request by tehilar**
some changes on OttSessionProvider

### CODE REVIEW

File: playkit/src/main/java/com/kaltura/playkit/backend/phoenix/data/OttResultAdapter.java [hunk 1]
```diff
@@ -32,9 +35,32 @@ public BaseResult deserialize(JsonElement json, Type typeOfT, JsonDeserializatio
...
+            ErrorElement error = new Gson().fromJson(result.get("error"), ErrorElement.class);
+
+            // trying to find constructor that excepts ErrorElement in order to return an object of type "typeOfT" even on error.
+            try {
+                Constructor<? extends Type> constructor = typeOfT.getClass().getConstructor(ErrorElement.class);
+
+                if (constructor != null) {
+                    baseResult = (BaseResult) constructor.newInstance(error);
+                }
+            } catch (NoSuchMethodException e) {
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/backend/phoenix/data/OttResultAdapter.java]: what's up with the empty catches? should probably unify them and add a comment saying why you ignore the error.

File: playkit/src/main/java/com/kaltura/playkit/backend/phoenix/OttSessionProvider.java [hunk 2]
```diff
@@ -386,26 +448,27 @@ public void clearSession() {
...
         refreshDelta = TimeDelta;
 
-        clearScheduled();
+        cancelCurrentRefreshTask();
     }
 
-    private boolean isScheduled() {
-        return scheduledRefreshTask != null && !scheduledRefreshTask.isDone() && !scheduledRefreshTask.isCancelled();
-    }
+    /*private boolean hasActiveRefreshTask() {
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/backend/phoenix/OttSessionProvider.java]: why commented-out?

File: playkit/src/main/java/com/kaltura/playkit/backend/phoenix/OttSessionProvider.java [hunk 3]
```diff
@@ -386,26 +448,27 @@ public void clearSession() {
...
+        return refreshTaskFurure != null && !refreshTaskFurure.isDone() && !refreshTaskFurure.isCancelled();
+    }*/
 
     @Override
     protected void setSession(String sessionToken, long expiry, String userId) {
         super.setSession(sessionToken, expiry, userId);
-        long delay = 1;
+        //long delay = 1;
```
Comment (noamtamim) [inline playkit/src/main/java/com/kaltura/playkit/backend/phoenix/OttSessionProvider.java]: why commented-out (and not removed)?

File: playkit/src/main/java/com/kaltura/playkit/backend/phoenix/OttSessionProvider.java (hunk 2 above)
Comment (tehilar) [inline playkit/src/main/java/com/kaltura/playkit/backend/phoenix/OttSessionProvider.java]: not in use

File: playkit/src/main/java/com/kaltura/playkit/backend/phoenix/OttSessionProvider.java (hunk 3 above)
Comment (tehilar) [inline playkit/src/main/java/com/kaltura/playkit/backend/phoenix/OttSessionProvider.java]: timer was removed

File: playkit/src/main/java/com/kaltura/playkit/backend/phoenix/data/OttResultAdapter.java (hunk 1 above)
Comment (tehilar) [inline playkit/src/main/java/com/kaltura/playkit/backend/phoenix/data/OttResultAdapter.java]: i'm trying to create an object of type typeOfT with ErrorElement receiving constructor.
but not all classes ha this constructor so the base class will return.
that why i have empty catches, if the constructor not exists - do nothing

**Pull request merged**